template/template_test.go

```diff
@@ -153,6 +154,45 @@ func TestTemplateExpansion(t *testing.T) {
 				}},
 			output: "a:11: b:21: ",
 		},
+		{
+			// Missing value is no value for nil options
+			text:   "{{ .Foo }}",
+			output: "<no value>",
+		},
+		{
+			// Missing value is no value for no options
+			text:    "{{ .Foo }}",
+			options: make([]string, 0),
+			output:  "<no value>",
+		},
+		{
+			// assert that missing value returns error with missingkey=error
+			text:       "{{ .Foo }}",
+			options:    []string{"missingkey=error"},
+			shouldFail: true,
+			errorMsg:   `error executing template test: template: test:1:3: executing "test" at <.Foo>: nil data; no entry for key "Foo"`,
+		},
+		{
+			// Missing value is "" for nil options in ExpandHTML
+			text:   "{{ .Foo }}",
+			output: "",
+			html:   true,
+		},
+		{
+			// Missing value is "" for no options in ExpandHTML
+			text:    "{{ .Foo }}",
+			options: make([]string, 0),
+			output:  "",
+			html:    true,
+		},
+		{
+			// assert that missing value returns error with missingkey=error in ExpandHTML
```

Comment:
We follow the comment style of starting with a capital letter (unless some keyword) and end with a full stop. Could you also update other comments here to follow that style?
```suggestion
			// Assert that missing value returns error with missingkey=error in ExpandHTML.
```